shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/surveyselection/SurveySelectionViewModel.kt

```diff
@@ -42,7 +52,41 @@ class SurveySelectionViewModel(
                 .combine(getAppVersion()) { userDate, version ->
                     Triple(userDate.first, userDate.second, version)
                 }
-                .collect { updateHeaderState(it.first, it.second, it.third) }
+                .combine(fetchSurvey(currentPage)) { userData, survey ->
+                    Pair(userData, survey)
+                }
+                .collect {
+                    updateHeaderState(it.first.first, it.first.second, it.first.third)
+                    updateSurveyState(it.second)
+                }
+        }
+    }
+
+    fun checkFetchMore(item: Int) {
+        if(item >= viewState.value.surveys.size - FETCH_MORE_TRIGGER) {
+            fetchMoreSurvey()
+        }
+    }
+
+    private fun fetchSurvey(page: Int): Flow<List<SurveyUiModel>> {
+        return flow {
+            surveyListUseCase(page)
+                .catch { emit(listOf()) }
```

Comment:
@blyscuit So, in case the current page is not able to fetch (lost network connection, for example), we will skip loading the current page and go to the next page? 🤔

Reply:
If the use case returns error then the next `fetchSurvey` will call the same page, only when data is returned that the page increases.